### ISSUE TITLE
Identify protocol to apply when the terms of use change

### ISSUE THREAD

**Issue by tschaffter** [nlpsandbox/nlpsandbox.io#14]
There are two reasons the rules may change while the project is active:

- This is a continuous benchmarking project, that is, it has currently no end date
- We are working to onboard new data sites. These sites may required NLP Sandbox user to agree to new use terms.

What has been do in past challenges in this situation:

- Empty list of registered participants
- Ask them to re-register

An issue with this solution is that once the list of registered participants is empty, this is lost information (there's no way to copy a team I think as this would required user to accept to join the new team). Renaming the team, for example with an "archived" suffix may not be a satisfying solution.

An alternative solution could be to work with teams named "NLP Sandbox Terms of use YYYY-MM-DD", in addition to a global team named "NLP Sandbox Users"? The former would be used for Files and Submission sharing permissions, while the latter would be keep a global list of all users and provide the mailing list address.

Question: Can the click wrap add a user to two teams (e.g. "NLP Sandbox User" and "NLP Sandbox Terms of use YYYY-MM-DD") and if yes, without throwing an error if the user is already in "NLP Sandbox User"?

**Comment by tschaffter** [comment]
@thomasyu888 Do you have insights on how the clickwrap works?

**Comment by thomasyu888** [comment]
@tschaffter I don't think so.  A click wrap is added on a team specific level, so you can add the same click wrap to two different teams, but people would still have to accept the terms twice...  I don't really like the idea of having multiple teams per new click wrap.

Ideally if new access controls are added to a team or Entity, people should have to re-accept the terms of use... Not sure how difficult that would be to do for the platform team.

**Comment by tschaffter** [comment]
Maybe 1 clickwrap to end up in the global user team + N * 1 clickwrap to be added to a team like "NLP Sandbox Terms Of Use YYYY-MM-DD"

where N is the number of times we set/change the terms of use. Permission would be given only to the member of the latest "NLP Sandbox Terms Of Use YYYY-MM-DD" team.

**Comment by thomasyu888** [comment]
Im going to post this use case in jira... because us having to manage permissions for N number of teams could be a logistical nightmare.

But agreed with the current solution.

**Comment by tschaffter** [comment]
Or actually we can stick with one team "NLP Sandbox Agreed to Terms Of Use"  that we empty when we update the terms of use. It's OK to empty this team because we would now have the other, global team that include all users who have ever registered.

**Comment by thomasyu888** [comment]
I like that solution better: https://sagebionetworks.jira.com/browse/PLFM-6590.   We could essentially mask the "Register for a challenge" button with. "Accept the NLP sandbox terms of use".  Every time there are new terms, then people would have to accept to the new terms.
